### PR TITLE
Ports: fix gcc for mac users

### PR DESCRIPTION
**Pull request by yegor-github**
Credit goes to matthewbjones#8780 on Discord.

### CODE REVIEW

**Comment by awesomekling** [comment]
I don't understand this patch. Can you please explain what was wrong, and how this fixes it?

**Comment by trflynn89** [comment]
On macOS, `sed` requires a file extension argument when using the `-i` flag - it's an extension appended to the filename to create a backup copy before editing the file. It's optional on Linux.

Came here to say I don't think this works as-is on Linux though. At least on my machine, there cannot be a space after the `-i` flag:
```bash
$ sed -i '' 's/abc/def/g' tmp
sed: can't read s/abc/def/g: No such file or directory

$ sed -i'' 's/abc/def/g' tmp 
```

**Comment by yegor-github** [comment]
For a portable solution I think it would have to change to `sed 's@-fno-exceptions @@' gcc/config/serenity.h > gcc/config/serenity.h.bak && mv gcc/config/serenity.h.bak gcc/config/serenity.h`.

**Comment by gunnarbeutner** [comment]
Here's how I fixed it for the toolchain: c60f02fbc2

**Comment by yegor-github** [comment]
Il do `sed -i='' 's@-fno-exceptions @@' gcc/config/serenity.h` then to keep consistency

**Comment by gunnarbeutner** [comment]
Also, you can update an existing PR by force-pushing the branch. However, the GitHub web UI doesn't support that so you might want to clone the repository and work on your branch from there. :)

**Comment by trflynn89** [comment]
Hm, on my machine, `sed -i=''` creates a backup with an extension of `=`:

```bash
$ ls
tmp

$ cat tmp 
abc def abc

$ sed -i='' 's/abc/def/g' tmp

$ ls
tmp  'tmp='

$ cat tmp
def def def

$ cat tmp= 
abc def abc
```

**Comment by yegor-github** [comment]
guess `sed -i.bak 's@-fno-exceptions @@' gcc/config/serenity.h && rm gcc/config/serenity.h.bak` works better then

**Comment by gunnarbeutner** [comment]
> Hm, on my machine, `sed -i=''` creates a backup with an extension of `=`:
> 

You're right. I guess you're the first to notice that and the toolchain has been creating those files all along. :)

**Comment by trflynn89** [comment]
Ha, yep!
```bash
$ find . -name '*='
./Toolchain/Tarballs/gcc-10.3.0/gcc/config/serenity.h=
./Toolchain/Tarballs/gcc-11.1.0/gcc/config/serenity.h=
```

@yegor-github Per [contributing guidelines](https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md#code-submission-policy), please squash your updates into the original commit. Thanks!
> Squash your commits when making revisions after a patch review.

**Comment by yegor-github** [comment]
> Ha, yep!
> 
> ```shell
> $ find . -name '*='
> ./Toolchain/Tarballs/gcc-10.3.0/gcc/config/serenity.h=
> ./Toolchain/Tarballs/gcc-11.1.0/gcc/config/serenity.h=
> ```
> 
> @yegor-github Per [contributing guidelines](https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md#code-submission-policy), please squash your updates into the original commit. Thanks!
> 
> > Squash your commits when making revisions after a patch review.

Hi, squashing commits is done with `git rebase -i HEAD~[number of commits]` right?

**Comment by trflynn89** [comment]
Yep - then you'll have to force-push to GitHub after the rebase.

**Comment by gunnarbeutner** [comment]
The latest change doesn't work for me:

```
$ ./package.sh
Installing dependencies of gcc!
Fetching gcc!
Downloading URL: https://ftpmirror.gnu.org/gnu/gcc/gcc-11.1.0/gcc-11.1.0.tar.xz
gcc-11.1.0.tar.xz already exists
Expecting sha256sum: 4c4a6fb8a8396059241c2e674b85b351c26a5d678274007f076957afa1cc9ddf
sha256sum(gcc-11.1.0.tar.xz) = '4c4a6fb8a8396059241c2e674b85b351c26a5d678274007f076957afa1cc9ddf'
+ tar -xf gcc-11.1.0.tar.xz (nocd)
+ touch .gcc-11.1.0.tar.xz_extracted
+ contrib/download_prerequisites
2021-06-04 21:32:40 URL:http://gcc.gnu.org/pub/gcc/infrastructure/gmp-6.1.0.tar.bz2 [2383840/2383840] -> "gmp-6.1.0.tar.bz2" [1]
2021-06-04 21:32:41 URL:http://gcc.gnu.org/pub/gcc/infrastructure/mpfr-3.1.4.tar.bz2 [1279284/1279284] -> "mpfr-3.1.4.tar.bz2" [1]
2021-06-04 21:32:42 URL:http://gcc.gnu.org/pub/gcc/infrastructure/mpc-1.0.3.tar.gz [669925/669925] -> "mpc-1.0.3.tar.gz" [1]
2021-06-04 21:32:43 URL:http://gcc.gnu.org/pub/gcc/infrastructure/isl-0.18.tar.bz2 [1658291/1658291] -> "isl-0.18.tar.bz2" [1]
gmp-6.1.0.tar.bz2: OK
mpfr-3.1.4.tar.bz2: OK
mpc-1.0.3.tar.gz: OK
isl-0.18.tar.bz2: OK
All prerequisites downloaded successfully.
Patching gcc!
+ patch -p1
patching file isl/config.sub
File mpc/config.sub is read-only; trying to patch anyway
patching file mpc/config.sub
patching file mpfr/config.sub
patching file gmp/configfsf.sub
+ touch .dependencies-config.patch_applied
+ patch -p1
patching file libstdc++-v3/include/c_compatibility/fenv.h
+ touch .fenv.patch_applied
+ patch -p1
patching file config.sub
patching file fixincludes/mkfixinc.sh
patching file gcc/config/arm/serenity-elf.h
patching file gcc/config/serenity.h
patching file gcc/config/serenity.opt
patching file gcc/config.gcc
patching file gcc/configure
patching file libgcc/config/t-slibgcc
patching file libgcc/config.host
patching file libgcc/unwind-dw2-fde-dip.c
patching file libstdc++-v3/configure
patching file libstdc++-v3/crossconfig.m4
patching file libstdc++-v3/configure
+ touch .gcc.patch_applied
Configuring gcc!
sed: can't read gcc/config/serenity.h: No such file or directory
```

**Comment by yegor-github** [comment]
> The latest change doesn't work for me:
> 
> ```
> $ ./package.sh
> Installing dependencies of gcc!
> Fetching gcc!
> Downloading URL: https://ftpmirror.gnu.org/gnu/gcc/gcc-11.1.0/gcc-11.1.0.tar.xz
> gcc-11.1.0.tar.xz already exists
> Expecting sha256sum: 4c4a6fb8a8396059241c2e674b85b351c26a5d678274007f076957afa1cc9ddf
> sha256sum(gcc-11.1.0.tar.xz) = '4c4a6fb8a8396059241c2e674b85b351c26a5d678274007f076957afa1cc9ddf'
> + tar -xf gcc-11.1.0.tar.xz (nocd)
> + touch .gcc-11.1.0.tar.xz_extracted
> + contrib/download_prerequisites
> 2021-06-04 21:32:40 URL:http://gcc.gnu.org/pub/gcc/infrastructure/gmp-6.1.0.tar.bz2 [2383840/2383840] -> "gmp-6.1.0.tar.bz2" [1]
> 2021-06-04 21:32:41 URL:http://gcc.gnu.org/pub/gcc/infrastructure/mpfr-3.1.4.tar.bz2 [1279284/1279284] -> "mpfr-3.1.4.tar.bz2" [1]
> 2021-06-04 21:32:42 URL:http://gcc.gnu.org/pub/gcc/infrastructure/mpc-1.0.3.tar.gz [669925/669925] -> "mpc-1.0.3.tar.gz" [1]
> 2021-06-04 21:32:43 URL:http://gcc.gnu.org/pub/gcc/infrastructure/isl-0.18.tar.bz2 [1658291/1658291] -> "isl-0.18.tar.bz2" [1]
> gmp-6.1.0.tar.bz2: OK
> mpfr-3.1.4.tar.bz2: OK
> mpc-1.0.3.tar.gz: OK
> isl-0.18.tar.bz2: OK
> All prerequisites downloaded successfully.
> Patching gcc!
> + patch -p1
> patching file isl/config.sub
> File mpc/config.sub is read-only; trying to patch anyway
> patching file mpc/config.sub
> patching file mpfr/config.sub
> patching file gmp/configfsf.sub
> + touch .dependencies-config.patch_applied
> + patch -p1
> patching file libstdc++-v3/include/c_compatibility/fenv.h
> + touch .fenv.patch_applied
> + patch -p1
> patching file config.sub
> patching file fixincludes/mkfixinc.sh
> patching file gcc/config/arm/serenity-elf.h
> patching file gcc/config/serenity.h
> patching file gcc/config/serenity.opt
> patching file gcc/config.gcc
> patching file gcc/configure
> patching file libgcc/config/t-slibgcc
> patching file libgcc/config.host
> patching file libgcc/unwind-dw2-fde-dip.c
> patching file libstdc++-v3/configure
> patching file libstdc++-v3/crossconfig.m4
> patching file libstdc++-v3/configure
> + touch .gcc.patch_applied
> Configuring gcc!
> sed: can't read gcc/config/serenity.h: No such file or directory
> ```

try it now

File: Ports/gcc/package.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@ post_fetch() {
 
 pre_configure() {
     patch_internal
-    run sed -i 's@-fno-exceptions @@' gcc/config/serenity.h
+    sed -i.bak 's@-fno-exceptions @@' gcc-${version}/gcc/config/serenity.h && rm gcc-${version}/gcc/config/serenity.h.bak
```
Comment (gunnarbeutner) [inline Ports/gcc/package.sh:18]:
```suggestion
    run sed -i.bak 's@-fno-exceptions @@' gcc/config/serenity.h
    run rm -f gcc/config/serenity.h.bak
```

**Comment by gunnarbeutner** [comment]
Yeah, `run` changed into whatever `workdir` is set to.

File: Ports/gcc/package.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ post_fetch() {
 
 pre_configure() {
     patch_internal
-    run sed -i 's@-fno-exceptions @@' gcc/config/serenity.h
+    run sed -i.bak 's@-fno-exceptions @@' gcc/config/serenity.h && rm gcc/config/serenity.h.bak
```
Comment (gunnarbeutner) [inline Ports/gcc/package.sh:18]:
```suggestion
    run sed -i.bak 's@-fno-exceptions @@' gcc/config/serenity.h
    run rm -f gcc/config/serenity.h.bak
```

Comment (gunnarbeutner) [reply Ports/gcc/package.sh:18]: Right now `run` isn't applied to `rm` and the build fails:

```
+ touch .gcc.patch_applied
Configuring gcc!
+ sed -i.bak s@-fno-exceptions @@ gcc/config/serenity.h
rm: cannot remove 'gcc/config/serenity.h.bak': No such file or directory
```

**Review by gunnarbeutner — APPROVED** [review]
Builds just fine with exception support enabled. :partying_face:

**Pull request merged**